docs/docs/examples/accordion.mdx

```diff
@@ -0,0 +1,76 @@
+---
+id: Accordion
+sidebar_label: Accordion
+---
+
+Welcome to this code example that demonstrates the implementation of an accordion item. An accordion is an element which could be expanded and collapsed on click.
+
+The accordion item consists of a header or a button and a content section. When the header/button is clicked, the content beneath it will smoothly expand or collapse, providing a clean and user-friendly way to manage information.
+
+One important thing to note is that when using `SharedValue` to handle state, we avoid unnecessary rerenders, and the work is handled by the animated style.
+
+Let's dive into the code to see how it works and how you can customize it for your specific needs.
+
+```tsx
+import {
+  View,
+  ViewProps
+} from "react-native";
+import Animated, {
+  SharedValue,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withTiming
+} from "react-native-reanimated";
+
+type AccordionProps = ViewProps & {
+  isExpanded: SharedValue<boolean>;
+  viewKey: number | string;
+  duration?: number;
+};
+
+const AccordionItem = ({
+  isExpanded,
+  children,
+  viewKey,
+  style,
+  duration = 500,
+}: AccordionProps) => {
+  const height = useSharedValue(0);
+
+  const derivedHeight = useDerivedValue(() =>
+    withTiming(height.value * Number(isExpanded.value), {
+      duration,
+    })
+  );
+  const bodyStyle = useAnimatedStyle(() => ({
+    height: derivedHeight.value,
+  }));
+
+  return (
+    <Animated.View
+      key={`accordionItem_${viewKey}`}
+      style={[
+        {
+          width: "100%",
+          overflow: "hidden",
+        },
+        bodyStyle,
+        style,
+      ]}
+    >
+      <View
+        onLayout={(e) => {
+          height.value = e.nativeEvent.layout.height;
+        }}
+        style={{ width: "100%", position: "absolute" }}
+      >
+        {children}
+      </View>
+    </Animated.View>
+  );
+};
+```
+
+See it live: https://snack.expo.dev/@majronman/accordion-item
```

Comment:
This is unsustainable for having snacks hosted on private accounts as it's unchangable by other developers 

The source code for snacks have to be hosted (preferably on github) and the link generated from our codebase


Some references how others have implemented it:
Expo docs: 
https://github.com/expo/expo/blob/main/docs/pages/tutorial/gestures.mdx#L173
https://github.com/expo/expo/blob/main/docs/ui/components/Snippet/blocks/SnackInline.tsx#L68
![image](https://github.com/software-mansion/react-native-reanimated/assets/39658211/0628cf57-9112-4b1d-870f-7cf8ee2871ff)

React Navigation:
![image](https://github.com/software-mansion/react-native-reanimated/assets/39658211/70caf030-f998-499c-b5ba-ec717f6a96e8)
https://github.com/react-navigation/react-navigation.github.io/blob/main/versioned_docs/version-6.x/navigating.md?plain=1#L31
https://github.com/react-navigation/react-navigation.github.io/blob/main/static/js/snack-helpers.js

There are different ways of implementing

As we have our source code open I hope to use GitHub as a CMS for our snacks